packages/cloud-config/yipfiles/01_defaults.yaml

```diff
@@ -50,124 +33,13 @@ stages:
          owner: 0
          group: 0
      - name: "Setup distro"
-       dns:
-         path: /etc/resolv.conf
-         nameservers:
-         - 8.8.8.8
-         - 1.1.1.1
        commands:
-       - ln -s /usr/share/zoneinfo/UTC /etc/localtime
-       - /usr/bin/systemctl enable cos-setup-reconcile.timer
-       - /usr/bin/systemctl enable cos-setup-fs-after.service
-       - /usr/bin/systemctl enable cos-setup-boot.service
-       - /usr/bin/systemctl enable cos-setup-network.service
-       - /usr/bin/systemctl mask purge-kernels
-       - mkdir /etc/ssl
-       - ln -s /var/lib/ca-certificates/pem /etc/ssl/certs
-       - ln -s /var/lib/ca-certificates/ca-bundle.pem /etc/ssl/ca-bundle.pem
+# commands always tries to expand vars (aka $..) so we can't pass a hashed passwd here
+#       - /bin/bash -c "/usr/sbin/usermod -p '$1$Oncxs0L7$U9/3Kuc2nP4oJowCNEquJ0' -g root root"
```

Comment:
I really struggled hours because of that... I'd say this might be a bug of the `yip` tool, I could not manage a way to run `/usr/sbin/usermod -p '$1$Oncxs0L7$U9/3Kuc2nP4oJowCNEquJ0' -g root root` and prevent variable expansion. Is there anything I missed about yip usage?

Reply:
Can be caused by `yip` templating, but either on how we run commands in `yip` - in both cases we rely on external libraries, so we might have to look at that closer

Reply:
this should be solved now, the new `yip` version has a `users` field that can be used to set passwords for users:

```yaml
stages:
 ...
     - name: "Setup users"
       users: 
          root: "$1$Oncxs0L7$U9/3Kuc2nP4oJowCNEquJ0"
```